### ISSUE TITLE
Plot Blank if Categorical Label too long

### ISSUE THREAD

**Issue by alanhdu** [bokeh/bokeh#4337]
If you create a figure with a categorical axis whose label is too long (I think the limit is around 50 characters, although that may be for my particular screen), then the plot shows up as blank. Panning and resizing do not show anything.

Ideally, either some kind of automatic truncation or some kind of error message would be great.

Sample code:

``` python
factors = ["a" * 100]
x = [50]

p = figure(y_range=factors)
p.circle(x, factors)
show(p)
```

``` bash
$ bokeh info
Python version      :  3.5.1 |Continuum Analytics, Inc.| (default, Dec  7 2015, 11:16:01) 
IPython version     :  4.2.0
Bokeh version       :  0.11.1
BokehJS static path :  /home/alan/anaconda/envs/test/lib/python3.5/site-packages/bokeh/server/static
```

**Comment by alanhdu** [comment]
http://nbviewer.jupyter.org/gist/alanhdu/274b09179f2b5ee2abfec0611a14faba is an example of what I'm talking about.

**Comment by birdsarah** [comment]
Thanks for the report @alanhdu. 

Have confirmed this. Also, this is not fixed by the main layout work, but given the status of the new layout-engine it should be relatively easy to find better constraints that accomodate this problem.

The root of the problem is that the axis width dominates and causes the plot to be tiny. When the axis is too big, there's no room for the plot.

In my tests the result is even uglier than the reference above.

**Comment by birdsarah** [comment]
FWIW I'm guessing this isn't just categorical labels.

**Comment by birdsarah** [comment]
Just to note this was *not* fixed by #4343

**Comment by bryevdv** [comment]
with the following code:
```
from bokeh.plotting import figure, show, output_file
factors = ["a" * 100]
x = [50]

p = figure(y_range=factors, plot_width=1000)
p.circle(x, factors)
output_file("bar.html")
show(p)
```
I now get

<img width="891" alt="screen shot 2017-08-02 at 22 23 30" src="https://user-images.githubusercontent.com/1078448/28904535-4af00a7e-77d1-11e7-8cbb-e6be8aead762.png">

If the plot width is reduced, then eventually you will see an "Unsatisfiable Contraint" error show up. But there is nothing Bokeh can do about that? If the plot is 1000px wide and a label is 1050px wide, that is not a problem we can do anything about. So, I am going to close this issue.
